service/history/transferQueueProcessor.go

```diff
@@ -455,16 +456,30 @@ func (t *transferQueueProcessorImpl) processCloseExecution(task *persistence.Tra
 		return err
 	}
 
+	isChildWorkflow := msBuilder.hasParentExecution()
+	workflowCloseStatus := msBuilder.executionInfo.CloseStatus
+	completionEvent, _ := msBuilder.GetCompletionEvent()
```

Comment:
We will unnecessarily deserialize the completionEvent even for execution which does not have a Parent.  Can you bring this back with the If check?

Reply:
sure